src/ArrayUtils.f90

```diff
@@ -202,7 +206,7 @@ PURE SUBROUTINE getDelta_1DReal(r,rout,xi)
 !> @param r The input array of reals
 !> @param delta The optional input for whether the array is incremental or not
 !> @param tol The tolerance for comparing two real values
-!> @param sout The number of unique entries in the array r.
+!> @returns sout The number of unique entries in the array r.
 !>
 FUNCTION findNUnique_1DReal(r,delta,tol) RESULT(sout)
```

Comment:
you didn't do this, but why the name sout?  Seems like something more descriptive would be better